hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -634,7 +632,7 @@ private static boolean areTablesToBeUpdated(UnresolvedEntityInsertActions action
 	}
 
 	private static String[] convertTimestampSpaces(Set spaces) {
-		return (String[]) spaces.toArray( new String[ spaces.size() ] );
+		return (String[]) spaces.toArray( new String[0] );
```

Comment:
As well known, modern JVM has better performance using zero length string as the parameter of `toArray()`.

Reply:
you're right but the difference is extremely tiny, not worth doing in 5.

Reply:
I reverted it back, together with many other trivial changes.